### ISSUE TITLE
Error on http://www.freeworldgroup.com/games6/gameindex/math-attack.htm

### ISSUE THREAD

**Issue by freeworldgroup** [ruffle-rs/ruffle#2674]
# Error Info
Error name: Error
Error message: panicked at 'already borrowed: BorrowMutError', /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/gc-arena-0.2.0/src/gc_cell.rs:62:29
Error stack:
```
Error: panicked at 'already borrowed: BorrowMutError', /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/gc-arena-0.2.0/src/gc_cell.rs:62:29
    at a.wbg.__wbg_new_df6e6ab7a65c4c4d (http://www.freeworldgroup.com/ruffle/ruffle.js:1:6333)
    at std::sync::once::Once::call_once::{{closure}}::h7141d078b1ea013f (http://www.freeworldgroup.com/ruffle/f9476b69ff64ab5157a65e17911e04b4.wasm:wasm-function[1270]:0x262d6b)
    at std::sync::once::Once::call_inner::h02eac160d339afc2 (http://www.freeworldgroup.com/ruffle/f9476b69ff64ab5157a65e17911e04b4.wasm:wasm-function[1537]:0x293cac)
    at core::ops::function::FnOnce::call_once{{vtable.shim}}::hf48d3d3ee3d10801 (http://www.freeworldgroup.com/ruffle/f9476b69ff64ab5157a65e17911e04b4.wasm:wasm-function[4313]:0x368b5f)
    at std::panicking::rust_panic_with_hook::h123718ba3bf480af (http://www.freeworldgroup.com/ruffle/f9476b69ff64ab5157a65e17911e04b4.wasm:wasm-function[3306]:0x341a6d)
    at std::panicking::begin_panic_handler::{{closure}}::hf393a82e58397bd2 (http://www.freeworldgroup.com/ruffle/f9476b69ff64ab5157a65e17911e04b4.wasm:wasm-function[3920]:0x35e9b1)
    at std::sys_common::backtrace::__rust_end_short_backtrace::h2ff2cfc953878925 (http://www.freeworldgroup.com/ruffle/f9476b69ff64ab5157a65e17911e04b4.wasm:wasm-function[4656]:0x36e1f9)
    at rust_begin_unwind (http://www.freeworldgroup.com/ruffle/f9476b69ff64ab5157a65e17911e04b4.wasm:wasm-function[4530]:0x36c879)
    at core::panicking::panic_fmt::h285dc015a87be5a4 (http://www.freeworldgroup.com/ruffle/f9476b69ff64ab5157a65e17911e04b4.wasm:wasm-function[4658]:0x36e25d)
    at core::option::expect_none_failed::hb95d2b1dc5448c17 (http://www.freeworldgroup.com/ruffle/f9476b69ff64ab5157a65e17911e04b4.wasm:wasm-function[4001]:0x36125c)
```

# Player Info
Allows script access: true

# Page Info
Page URL: http://www.freeworldgroup.com/games6/gameindex/math-attack.htm
SWF URL: http://www.freeworldgroup.com/games6/mathattack/mathAttack.swf

# Browser Info
Useragent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/87.0.4280.141 Safari/537.36
OS: Win32

# Ruffle Info
Version: 0.1.0
Name: nightly 2021-01-20
Channel: nightly
Built: 2021-01-20T01:07:01.467Z
Commit: 75967e377576cad9a679f1172d0b7f6e5ffd85c0

**Comment by Lord-McSweeney** [comment]
No more panic, but it looks like this is stuck on serialization for other objects not yet having been implemented: `Serialization is not implemented for class other than Object: ScriptObject(ScriptObject { name: FWG::PlayerInfo, ptr: 0x42c208 })`. This leads to a cascade of errors.

**Comment by Sonikkustar** [comment]
Game appears to work now without any immediate issues apparent.

![image](https://github.com/ruffle-rs/ruffle/assets/22226938/34dea3ed-f4d4-460f-998a-c7dcf9fc8c87)

Tested with **nightly-2023-12-15**

**Comment by n0samu** [comment]
Likely fixed by #11971
